lib/SILOptimizer/Mandatory/TFDifferentiation.cpp

```diff
@@ -1096,9 +1096,6 @@ class ADContext {
   DifferentiationTask *
   lookUpDifferentiationTask(SILFunction *original,
                             const SILReverseAutoDiffIndices &indices) {
-    auto *attr = lookupReverseDifferentiableAttr(original, indices);
```

Comment:
Why is it no longer looking up the attribute?

Reply:
Each differentiation task is associated with a `[reverse_differentiable]` attr. If the attr does not exist but an enqueued task can be found, there must be an error.

Reply:
This lookup and the `if (!attr) return nullptr;` don't ever affect anything because the attr will always be there if a task is there.

(I originally thought this was a bug because it seemed odd for a function that's looking up a differentiation task to exit early if there isn't an attr, because the function shouldn't care whether there is an attr or not -- all it cares about is whether there is a task. Upon closer inspection, this code doesn't actually do anything because the attr will always be there when a task is there.)

Reply:
So yes, this change is not actually necessary to fix the problem. But it seems like a good change anyways because it simplifies the function.

Reply:
I think I'd be nicer to make it an assertion when a task is found. But I don't want to waste another cycle of CI :) Could you add it whenever you change AD next time?

Reply:
ok